test/test_check_db_dump_recent.py

```diff
@@ -0,0 +1,276 @@
+#!/usr/bin/python3
+
+"""
+This file tests the script that checks the GOCDB failover process is
+happening.
+"""
+
+from datetime import datetime, timedelta
+import os
+import tempfile
+import unittest
+
+# Need to import the check_db_dump_recent module as a whole to monkey patch
+# the underlying datetime.now() calls.
+from check import check_db_dump_recent
+
+
+class TestCheckDBDumpRecent(unittest.TestCase):
+    """
+    This class tests the check.check_db_dump_recent module.
+
+    For ORACLE test cases, the example log format is as follows:
+        YYYY-MM-DDTHH:MM:SS+0000
+        YYYY-MM-DDTHH:MM:SS+0000 INFO: completed ok
+        YYYY-MM-DDTHH:MM:SS+0000
+        An Error
+        YYYY-MM-DDTHH:MM:SS+0000
+        2023-01-01T11:41:01+0000 INFO: completed ok
+
+    The example log formats follow the style of the actual logs generated.
+    That is to say, an attempt to restore the database dump:
+      - Starts with an ISO 8601 timestamp, this is because the restore is
+      triggered by a cron job based off quattor cron jobs.
+      - Ends with either (generated from the 1_runDbUpdate.sh):
+        - an error with no timestamp
+        - an ISO 8601 timestamp and the string "INFO: completed ok"
+
+    For MariaDB test cases, the example log format is as follows:
+        YYYY-MM-DDTHH:MM:SS+0000
+        YYYY-MM-DDTHH:MM:SS+0000 INFO: completed ok
+        YYYY-MM-DDTHH:MM:SS+0000
+        YYYY-MM-DDTHH:MM:SS+0000 ERROR: An Error
+        YYYY-MM-DDTHH:MM:SS+0000
+        YYYY-MM-DDTHH:MM:SS+0000 INFO: completed ok
+
+    The example log formats follow the style of the actual logs generated.
+    That is to say, an attempt to restore the database dump:
+      - Starts with an ISO 8601 timestamp, this is because the restore is
+      triggered by a cron job based off quattor cron jobs.
+      - Ends with an ISO 8601 timestamp followed by either (generated by
+        failover_import.py):
+        - an error
+        - the string "INFO: completed ok"
+    """
+
+    def setUp(self):
+        self.test_checker = check_db_dump_recent.CheckDBDumpRecent(
+            grace_period=timedelta(
+                minutes=30
+            )
+        )
+
+        # Create and close a temporary log file for use in the tests.
+        temp_file, self.temp_file_path = tempfile.mkstemp()
+        os.close(temp_file)
+
+        # Monkey patch the datetime module within check.check_db_dump_recent
+        # so we don't have to worry about hard coded datetimes in these tests.
+        check_db_dump_recent.datetime = MonkeyDateTime
+        help(check_db_dump_recent)
```

Comment:
What's the point of invoking `help` here?

Reply:
ha, no point, i'll remove.